core/src/main/scala/org/apache/spark/status/api/v1/ApplicationListResource.scala

```diff
@@ -42,7 +44,7 @@ private[v1] class ApplicationListResource(uiRoot: UIRoot) {
     }
     val includeCompleted = adjStatus.contains(ApplicationStatus.COMPLETED)
     val includeRunning = adjStatus.contains(ApplicationStatus.RUNNING)
-    val appList = allApps.filter { app =>
+    allApps.filter { app =>
```

Comment:
So I've just realized a fundamental issue with this code change. This filter is being done *after* the limit now. This mean if you limit to say 100 app but set the date to a year ago you'd expect 100 apps from a year ago, but this instead will return apps from a year ago from the last 100 apps, which for a busy server would be 0. 
This means the limit *must* be run after this filter.